### ISSUE TITLE
ensurePackage don't work with @nx/angular

### ISSUE THREAD

**Issue by DominikPieper** [nrwl/nx#16696]
### Current Behavior

I have a generator using the Angular app generator from @nx/angular. I add @nx/angular through the devkit function "ensurePackage". Adding other Nx packages worked well; with v15 @nrwl/angular, it was working as expected.
I tried different tsconfig options but couldn't find the one doing the trick.

One module of the @nx/angular package uses esnext, is that the reason?

### Expected Behavior

Generator can run

### GitHub Repo

https://github.com/nxext/nx-extensions/blob/update_nx_16/packages/ionic-angular/src/generators/application/lib/add-angular.ts

### Steps to Reproduce

1. Check out Nxext repo
2. Remove ionic-angular-e2e from .nxignore
3. Run the e2e test for ionic-angular

(Cloud run https://cloud.nx.app/runs/xKHVcwp700)

### Nx Report

```shell
Node   : 18.13.0
   OS     : darwin x64
   yarn   : 1.22.15
   Hasher : Native

   nx                 : 16.0.1
   @nx/js             : 16.0.1
   @nx/jest           : 16.0.1
   @nx/linter         : 16.0.1
   @nx/workspace      : 16.0.1
   @nx/angular        : 16.0.1
   @nx/cypress        : 16.0.1
   @nx/devkit         : 16.0.1
   @nx/eslint-plugin  : 16.0.1
   @nx/node           : 16.0.1
   @nx/plugin         : 16.0.1
   @nx/react          : 16.0.1
   @nx/storybook      : 16.0.1
   @nrwl/tao          : 16.0.1
   @nx/vite           : 16.0.1
   @nx/web            : 16.0.1
   @nx/webpack        : 16.0.1
   nx-cloud           : 16.0.5
   typescript         : 4.9.5
   ---------------------------------------
   Local workspace plugins:
   	 @nxext/ionic-angular
   	 @nxext/ionic-react
   	 @nxext/capacitor
   	 @nxext/preact
   	 @nxext/svelte
   	 @nxext/vitest
   	 @nxext/vite
   	 @nxext/vue
   	 @nxext/e2e
```

### Failure Logs

```shell
require() of ES Module ~/nxext/node_modules/@nx/angular/fesm2015/nx-angular.mjs not supported.

   Instead change the require of ~/nxext/node_modules/@nx/angular/fesm2015/nx-angular.mjs to a dynamic import() which is available in all CommonJS modules.

Error [ERR_REQUIRE_ESM]: require() of ES Module ~/nxext/node_modules/@nx/angular/fesm2015/nx-angular.mjs not supported.
Instead change the require of ~/nxext/node_modules/@nx/angular/fesm2015/nx-angular.mjs to a dynamic import() which is available in all CommonJS modules.
    at Mod.require (~//nxext/tmp/nx-e2e/proj/node_modules/nx/bin/init-local.js:204:36)
    at require (~/nxext/tmp/nx-e2e/proj/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at ensurePackage (~/nxext/tmp/nx-e2e/proj/node_modules/@nx/devkit/src/utils/package-json.js:249:16)
    at ~/nxext/tmp/nx-e2e/proj/node_modules/@nxext/ionic-angular/src/generators/application/lib/add-angular.js:10:36
    at Generator.next (<anonymous>)
    at ~/nxext/tmp/nx-e2e/proj/node_modules/tslib/tslib.js:167:75
    at new Promise (<anonymous>)
    at Object.__awaiter (~/nxext/tmp/nx-e2e/proj/node_modules/tslib/tslib.js:163:16)
    at addAngular (~/nxext/tmp/nx-e2e/proj/node_modules/@nxext/ionic-angular/src/generators/application/lib/add-angular.js:9:20)
    at ~/nxext/tmp/nx-e2e/proj/node_modules/@nxext/ionic-angular/src/generators/application/generator.js:17:64
    at Generator.next (<anonymous>)
    at ~/nxext/tmp/nx-e2e/proj/node_modules/tslib/tslib.js:167:75
    at new Promise (<anonymous>)
    at Object.__awaiter (~/nxext/tmp/nx-e2e/proj/node_modules/tslib/tslib.js:163:16)
    at applicationGenerator (~/nxext/tmp/nx-e2e/proj/node_modules/@nxext/ionic-angular/src/generators/application/generator.js:14:20)
    at Object.<anonymous> ~/nxext/tmp/nx-e2e/proj/node_modules/nx/src/command-line/generate.js:249:36)
    at Generator.next (<anonymous>)
    at fulfilled (~/nxext/tmp/nx-e2e/proj/node_modules/tslib/tslib.js:164:62)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by hadrien-toma** [comment]
Same on my side... As my workflow is heavily based on this kind of generator composition, it breaks... everywhere... and especially for the `@nx/angular` one.

Here we meet again @DominikPieper haha :sweat_smile: :wave: :wink:

**Comment by DominikPieper** [comment]
@hadrien-toma  hey again 😁👋

all other packages I used so far worked not just @nx/angular broke for me

**Comment by nhhockeyplayer** [comment]
any chance of this being expedited and prioritized. 3rd quarter projects are trying to take off. Major players of Nx in the field are on hold waiting on this one. Thanks
